### PR TITLE
Improve the communication around required and optional

### LINKED ISSUE #35191: [Enhancement]: Improve the communication around required and optional fields
### Describe the solution you'd like

Given that the product name is the only required field in the whole form, we should mark it as such and remove the (optional) label from the `Summary` and `Sale price` fields (and any other that has it).

**Acceptance criteria** 

- [x] The `Name` field has a `(required)` suffix.
- [x] The `Summary` and `Sale price` fields no longer have the `(optional)` suffix.

### Describe alternatives you've considered

_No response_

### Additional context

image

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/35191.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go the Add new product page
2. The only field with `(required)` hint on the right side of its label should be the `Name` and the `(optional)` hit should not be shown for any field
3. This also apply to the New shipping class dialog form

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/shared/add-new-shipping-class-modal/add-new-shipping-class-modal.tsx [hunk 1]
```diff
@@ -39,13 +39,9 @@ function ShippingClassForm( { onAdd, onCancel }: ShippingClassFormProps ) {
...
 			<TextControl
 				{ ...getInputProps( 'name' ) }
-				label={ __( 'Name', 'woocommerce' ) }
-			/>
-			<TextControl
-				{ ...getInputProps( 'slug' ) }
 				label={ interpolateComponents( {
 					mixedString: __(
-						'Slug {{span}}(optional){{/span}}',
+						'Name {{span}}(required){{/span}}',
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/shared/add-new-shipping-class-modal/add-new-shipping-class-modal.tsx]: I wonder if passing just an `{{required/}}` component would make this slightly easier on translators.

E.g.,
```
components: {
  optional: {
    <span className="woocommerce-add-new-shipping-class-modal__optional-input">{ __( '(required)', 'woocommerce' ) }</span>
  }
}
```

**Review by joshuatf — DISMISSED** [review]
Looks great!  Left one optional change, but pre-approving!  ✅

File: plugins/woocommerce-admin/client/products/shared/add-new-shipping-class-modal/add-new-shipping-class-modal.tsx (hunk 1 above)
Comment (mdperez86) [reply plugins/woocommerce-admin/client/products/shared/add-new-shipping-class-modal/add-new-shipping-class-modal.tsx]: Cool!!! This is way better. Thanks and Done!!!

**Review by joshuatf — APPROVED** [review]
Beautiful!  Still testing well and LGTM 🚢

**Pull request merged**